source/MaterialXCore/Node.cpp

```diff
@@ -252,14 +252,21 @@ vector<ElementPtr> NodeGraph::topologicalSort() const
 
     const vector<ElementPtr>& children = getChildren();
 
-    // Calculate in-degrees for all nodes, and enqueue those with degree 0.
-    std::unordered_map<ElementPtr, size_t> inDegree(children.size());
+    // Calculate in-degrees for all children.
+    std::unordered_map<ElementPtr, int> inDegree(children.size());
```

Comment:
To my mind, the original `size_t` type in this map seemed more clear, since an in-degree can only be zero or positive.  Would it be reasonable to keep this as size_t, and then to adjust the decrement logic below to prevent negative values from being generated?